modules/internal/String.chpl

```diff
@@ -688,6 +688,50 @@ module String {
       return ret;
     }
 
+    // Checks to see if r is inside the bounds of this and returns a finite
+    // range that can be used to iterate over a section of the string.
+    // Converts from codePointIndex range to byte index range in the process.
+    pragma "no doc"
+    proc _getView(r:range(?)) where r.idxType == codePointIndex {
+      if boundsChecking {
+        if r.hasLowBound() {
+          if r.low:int <= 0 then
+            halt("range out of bounds of string");
```

Comment:
These should really use HaltWrappers but it looks like there's more to do in String.chpl along those lines.

Reply:
Thanks.  I didn't know about HaltWrappers; I just copied the byte-indexed version.  It does sound like HaltWrappers might be unevenly supported among modules.